### ISSUE TITLE
calicoctl patch: Label disappears after patching GlobalNetworkSet.

### ISSUE THREAD

**Issue by ira-gordin-sap** [projectcalico/calico#4724]
Label disappears after patching GlobalNetworkSet after running the following command: 
`calicoctl patch GlobalNetworkSet ip-protect -p '{"spec": {"nets": ["1.1.1.1","222.188.8.183", "223.25.247.152"]}}'`
## Expected Behavior

Label will stay in the place.

## Current Behavior

Label disappears.

## Steps to Reproduce (for bugs)

`calicoctl patch GlobalNetworkSet ip-protect -p '{"spec": {"nets": ["1.1.1.1","222.188.8.183", "223.25.247.152"]}}'`
===
Before:
```
apiVersion: projectcalico.org/v3
kind: GlobalNetworkSet
metadata:
  creationTimestamp: "2021-07-06T13:43:21Z"
  labels:
    IP-deny-list: "true"
  name: ip-protect
  resourceVersion: "2377838"
  uid: 8e150d95-81ac-4e27-b16d-5a1a47bd20e5
spec:
  nets:
  - 1.1.1.1
  - 1.157.168.58
  - 2.18.66.49
  - 2.18.66.56
  - 2.18.66.168
```
===
After:
```
apiVersion: projectcalico.org/v3
kind: GlobalNetworkSet
metadata:
  creationTimestamp: "2021-07-06T13:43:21Z"
  name: ip-protect
  resourceVersion: "2378250"
  uid: 8e150d95-81ac-4e27-b16d-5a1a47bd20e5
spec:
  nets:
  - 1.1.1.1
  - 222.188.8.183
  - 223.25.247.152
```
===
## Your Environment

* Calico version
* Orchestrator version (e.g. kubernetes, mesos, rkt):
kubernetes
* Operating System and version:
* Link to your project (optional):

**Comment by caseydavenport** [comment]
Yep, this definitely sounds like a bug to me. We'll take a look.

You might want to try using the Calico API server and `kubectl patch` as an alternative approach available in recent Calico releases.
